docs/WORKER_ARCHITECTURE.md

```
@@ -0,0 +1,220 @@
+
+```typescript
+interface SessionData {
+  userId?: string;
+  username?: string;
+  authenticated: boolean;
+  createdAt: number;
+  lastActivityAt: number;
+  preferences?: {
+    defaultApp?: 'console' | 'profile' | 'search';
+  };
+}
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)